### PR TITLE
gpu: move blocking pass from custom to advanced mode

### PR DESCRIPTION
**Pull request by italoaug**

### CODE REVIEW

**Comment by georgebisbas** [comment]
Hi, what is this PR offering compared to the previous implementation? Is loop blocking expected to deliver performance by default now?

File: devito/core/gpu.py [hunk 1]
```diff
@@ -185,6 +188,10 @@ def _specialize_iet(cls, graph, **kwargs):
         # Distributed-memory parallelism
         mpiize(graph, sregistry=sregistry, options=options)
 
+        # Loop tiling
+        if options['blocklevels']:
```
Comment (FabioLuporini) [inline devito/core/gpu.py:192]: does it fail without this?

**Review by FabioLuporini — COMMENTED** [review]
Minor question about the need for `if options[blocklevels] ...` but otherwise looking good :ok_hand: 

@georgebisbas : see latest discussion in `gpu-dev`

File: devito/core/gpu.py (hunk 1 above)
Comment (georgebisbas) [reply devito/core/gpu.py:192]: Should be fine to remove yes? Unless we miss sth.

File: tests/conftest.py [hunk 2]
```diff
@@ -311,14 +311,8 @@ def assert_blocking(operator, exp_nests):
...
-                       ('openacc', 'blocking', {'skewing': True}),
-                       ('openacc', 'blocking', {'skewing': True, 'blockinner': True})]
-
-
-# A list of optimization options/pipelines to be used in testing
-# regarding GPU openmp spatial and/or temporal blocking.
-opts_openmp_tiling = [('openmp', 'blocking'),
-                      ('openmp', 'blocking', {'skewing': True}),
-                      ('openmp', 'blocking', {'skewing': True, 'blockinner': True})]
+# regarding GPU spatial and/or temporal blocking.
```
Comment (georgebisbas) [inline tests/conftest.py:314]: If we ask for `blocking` only, do we get blocklevels=1 now?

File: devito/core/gpu.py (hunk 1 above)
Comment (italoaug) [reply devito/core/gpu.py:192]: You're right. It works fine without the conditional. So I'm removing it for the next commit.

**Review by FabioLuporini — APPROVED** [review]
LGTM, merging

**Pull request merged**